src/aspire/nfft/cufinufft.py

```diff
@@ -0,0 +1,131 @@
+from ctypes import c_int
+
+import logging
+import numpy as np
+import pycuda.autoinit
+import pycuda.driver as cuda
+import pycuda.gpuarray as gpuarray
+from cufinufft import cufinufft
+
+from aspire.nfft import Plan
+from aspire.utils import ensure
+
+logger = logging.getLogger(__name__)
+
+class cuFINufftPlan(Plan):
+    def __init__(self, sz, fourier_pts, epsilon=1e-15, many=0, **kwargs):
+        """
+        A plan for non-uniform FFT (3D)
+
+        :param sz: A tuple indicating the geometry of the signal
+        :param fourier_pts: The points in Fourier space where the Fourier transform is to be calculated,
+            arranged as a 3-by-K array. These need to be in the range [-pi, pi] in each dimension.
+        :param epsilon: The desired precision of the NUFFT
+        :param many: Optional integer indicating if you would like to compute a batch of `many`
+        transforms.  Implies vol_f.shape is (..., `many`). Defaults to 0 which disables batching.
+        """
+
+        # Passing "many" expects one large higher dimensional array.
+        #   Set some housekeeping variables so we can discern how to handle the dims later.
+        self.many = False
+        self.ntransforms = 1
+        if many != 0:
+            self.many = True
+            self.ntransforms = many
+
+        # Basic dtype passthough.
+        dtype = fourier_pts.dtype
+        if dtype == np.float64 or dtype == np.complex128:
```

Comment:
Recently we added `real_type` and `complex_type` functions in utils/misc.py to avoid using the if statement.

Reply:
as the person who wrote that function, you should know that this if statement is different than the one your wrote.

I did make use of the function elsewhere.